lib/components/base_component.dart

```diff
@@ -121,6 +118,11 @@ abstract class BaseComponent extends Component {
     if (gameRef is BaseGame) {
       gameRef.prepare(c);
     }
+
+    final childOnLoadFuture = c.onLoad();
+    if (childOnLoadFuture != null) {
+      await childOnLoadFuture;
```

Comment:
Do we really want to be blocking here? It could start loading several things if we weren't.

Reply:
This is up to the programmer, if they don't want for it to be blocking, they can just skip the `await` when calling `addChild`.

I think it is nice to give both options, as there may be cases where the programmer wants to be sure that the child was added and loaded

Reply:
How can they skip it, I think I'm missing something?
They only control the `childOnLoadFuture`, and that has an explicit `await` on it here?

Reply:
this `await` is called only if there is an `onLoad` future, so for the ones that returns `null`, this will no block.

For the cases where there is a Future, the programmer can just to wait for it or not, like:

```dart
await c.addChild(child);
```

or not wait

```dart
c.addChild(child);
```